recipes/liquid-dsp/all/conanfile.py

```diff
@@ -49,138 +46,81 @@ def _target_name(self):
 
     @property
     def _lib_pattern(self):
-        if self.settings.os == "Macos" and not self.options.shared:
+        if is_apple_os(self) and not self.options.shared:
             return "libliquid.a"
-        if self.settings.os != "Windows":
-            return self._target_name
-        return "libliquid.lib"
+        return self._target_name
 
-    def configure(self):
-        del self.settings.compiler.cppstd
-        del self.settings.compiler.libcxx
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
 
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
+    def configure(self):
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
 
-    def build_requirements(self):
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
-        if self.settings.compiler == "Visual Studio":
-            self.build_requires("mingw-w64/8.1")
-            self.build_requires("automake/1.16.4")
-            
-    def export_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def validate(self):
-        if hasattr(self, "settings_build") and tools.cross_building(self):
+        if cross_building(self):
             raise ConanInvalidConfiguration("Cross building is not yet supported. Contributions are welcome")
+        if is_msvc(self):
```

Comment:
In Conan 1 this recipe supports Windows, why are we removing support in this migration?

Reply:
It did so only through fetching MinGW via Conan and building with it instead of MSVC. I did try that approach and got it to work locally, but it kept failing on C3I due to a missing DLL despite at least 10 iterations of trying to fix this (and Windows is very unhelpful with its DLL error messages).
Relying on GCC from MinGW is a bit of a fragile edge-case for Conan, since it also becomes a build/runtime dependency unless you link GCC statically into a shared library. It's not well supported. See also https://github.com/conan-io/conan-center-index/pull/21193#issuecomment-2085602537

Reply:
Okey, we will leave this for the future then. It is better to have this recipe working in `conan v2` for now. Even if it does not support MSVC